src/writable_impl.rs

```diff
@@ -375,6 +375,34 @@ impl< C: Context, T: Writable< C > > Writable< C > for Option< T > {
     }
 }
 
+impl< C: Context, T: Writable< C >, E: Writable< C > > Writable< C > for Result< T, E > {
+    #[inline]
+    fn write_to< W: ?Sized + Writer< C > >( &self, writer: &mut W ) -> Result< (), C::Error > {
+        match *self {
+            Ok( ref value ) => {
+                writer.write_u8( 0 )?;
+                value.write_to( writer )
+            },
+            Err( ref value ) => {
+                writer.write_u8( 1 )?;
+                value.write_to( writer )
+            }
```

Comment:
Hm.... I think it'd be more natural to have it be `1` for `Ok` and `Err` to be `0`, since `Option::Some` is `1`.

(Or in other words, `Option::None` and `Result::Err(())` should serialize to the same thing.)

Reply:
Ok, I can switch the value so that `1` for `Ok` and `0` for `Err`